src/VolumeCache.ts

```diff
@@ -210,67 +145,29 @@ export default class VolumeCache {
 
-  /**
-   * Attempts to get data from a single channel. Internal implementation of `get`,
-   * which is overloaded to call this in different patterns.
-   */
-  private getOneChannel(
-    volume: CacheStore,
-    channel: number,
-    optDims: Partial<QueryExtent> = {}
-  ): ArrayBuffer | undefined {
-    // TODO allow searching through a range of scales and picking the highest available one
-    const scaleCache = volume.scales[optDims.scale || 0];
-    const entryList = scaleCache.data[optDims.time || 0][channel];
-    const chunk = optDims.chunk ?? new Vector3(0, 0, 0);
-
-    for (const entry of entryList) {
-      if (entry.chunk.equals(chunk)) {
-        this.moveEntryToFirst(entry);
-        return entry.data;
-      }
+  /** Internal implementation of `get`. Returns all entry metadata, not just the raw data. */
+  private getEntry(key: string): CacheEntry | undefined {
+    const result = this.entries.get(key);
+    if (result) {
+      this.moveEntryToFirst(result);
     }
-
-    return undefined;
+    return result;
   }
 
-  /** Attempts to get data from a single channel of a cached volume. Returns `undefined` if not present in the cache. */
-  public get(volume: CacheStore, channel: number, optDims?: Partial<QueryExtent>): ArrayBuffer | undefined;
-  /** Attempts to get data from multiple channels of a volume. Channels not present in the cache are `undefined`. */
-  public get(volume: CacheStore, channel: number[], optDims?: Partial<QueryExtent>): (ArrayBuffer | undefined)[];
-  /** Attempts to get all channels of a volume from the cache. Channels not present in the cache are `undefined`. */
-  public get(volume: CacheStore, optDims?: Partial<QueryExtent>): (ArrayBuffer | undefined)[];
-  public get(
-    volume: CacheStore,
-    channel?: number | number[] | Partial<QueryExtent>,
-    optDims?: Partial<QueryExtent>
-  ): ArrayBuffer | undefined | (ArrayBuffer | undefined)[] {
-    if (Array.isArray(channel)) {
-      return channel.map((c) => this.getOneChannel(volume, c, optDims));
-    }
-
-    if (typeof channel === "object" || channel === undefined) {
-      const channelKeys = [...Array(volume.numChannels).keys()];
-      return channelKeys.map((c) => this.getOneChannel(volume, c, channel));
-    }
-
-    return this.getOneChannel(volume, channel, optDims);
+  /** Attempt to get a single entry from the cache. */
+  public get(key: string): ArrayBuffer | undefined {
+    return this.getEntry(key)?.data;
   }
 
-  /** Clears data associated with one volume from the cache */
-  public clearVolume(volume: CacheStore): void {
-    volume.scales.forEach((scale) => {
-      scale.data.forEach((time) => {
-        time.forEach((channel) => {
-          channel.forEach((entry) => {
-            this.evict(entry);
-          });
-        });
-      });
-    });
+  public clearWithPrefix(prefix: string): void {
+    for (const [key, entry] of this.entries.entries()) {
+      if (key.startsWith(prefix)) {
+        this.evict(entry);
```

Comment:
Minor nit: I'm assuming speed is not a concern here.  But I wonder if it might be slightly more efficient to get all the key names first, and then do the deletions? something like: this.entries.keys.reduce(...startsWith...).map(evict(key))

Reply:
or, filter instead of reduce

Reply:
Is your concern over deleting from a collection while also iterating over it?

Reply:
Yes but it's not a significant concern, just musing about alternate ways.